### PR TITLE
#400 - fixed n+1 queries in app

### LINKED ISSUE #400: resolve N+1 queries
**Bug description**
> Sentry reports N+1 Query  problem.
`select * from "vacations" where "vacations"."vacation_request_id" = ? and "vacations"."vacation_request_id" is not null`
https://blumilk.sentry.io/issues/4868147428/?project=4505271823237120&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=30d&stream_index=0
Additionally there other models which are lazy loaded (Attempted to lazy load [roles] on model [Spatie\Permission\Models\Permission] but lazy loading is disabled.).

**Reproduction steps**
> Steps to reproduce the behavior:
> 1. Enable `Model::preventLazyLoading();` in `boot()` in AppServiceProvider
> 2. Login to the app

**Expected behavior**
> There should not be N+1 queries and Eloquent models should not be lazy loaded.

**Screenshots**
image
---
image

**Environment**
> - OS:
> - Browser:

### PR DESCRIPTION
**Pull request by kamilpiech97**
Should close #400.
Resolved `n+1` queries for all test firstly, then I clicked through all the pages and options throughout the application and found some `n+1` issues and fixed them.

@EwelinaSkrzypacz, u must click through all the pages again and test my changes.

### CODE REVIEW

**Comment by krzysztofrewak** [comment]
Does it solve #398 as well?

**Comment by kamilpiech97** [comment]
> Does it solve #398 as well?

I do not think so, I see that in telescope:
image

we will optimize dashboard in new one PR.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
I can't find more n+1 queries in the app.

But I noticed that we have a lot of duplicated queries (not only on dashboard #398) but for example on index request page


image


Maybe we can solve this in this task or in #398?

File: storage/debugbar/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*
```
Comment (krzysztofrewak) [inline storage/debugbar/.gitignore:1]: Please add debugbar to composer manifest.

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: storage/debugbar/.gitignore (hunk 1 above)
Comment (kamilpiech97) [reply storage/debugbar/.gitignore:1]: Added :D

**Review by krzysztofrewak — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**